activatesshkey/util.go

```diff
@@ -0,0 +1,176 @@
+package activatesshkey
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/bitrise-io/go-steputils/tools"
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/fileutil"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/pathutil"
+)
+
+func unsetEnvsBy(value string) error {
+	for _, env := range os.Environ() {
+		key, val := splitEnv(env)
+
+		if val == value {
+			if err := os.Unsetenv(key); err != nil {
+				return err
+			}
+
+			if err := tools.ExportEnvironmentWithEnvman(key, ""); err != nil {
+				return err
+			}
+			log.Debugf("%s has been unset", key)
+		}
+	}
+	return nil
+}
+
+func splitEnv(env string) (string, string) {
+	e := strings.Split(env, "=")
+	return e[0], strings.Join(e[1:], "=")
+}
+
+func ensureSavePath(savePath string) error {
+	dirpath := filepath.Dir(savePath)
+	return os.MkdirAll(dirpath, 0700)
```

Comment:
in the previous version, we used different permission: [0600](https://github.com/bitrise-steplib/steps-activate-ssh-key/blob/master/main.go#L69)
is this change intentional?

Reply:
Yes, if the directory did not exist previously, will not have permission to create files inside with the existing permissions.